include/sys/arch_interface.h

```diff
@@ -675,6 +675,21 @@ extern u64_t arch_timing_value_swap_temp;
 
 /** @} */
 
+/**
+ * @defgroup arch_cache Architecture-specific cache functions
+ * @ingroup arch-interface
+ * @{
+ */
+
+#ifdef CONFIG_CACHE_FLUSHING
+void arch_cache_flush(void *addr, size_t size, u32_t flags);
```

Comment:
This will also need documentation. Can be brief, with a "@see" reference to the APIs in cache.h.